sky/setup_files/dependencies.py

```diff
@@ -60,6 +60,7 @@
     'psycopg2-binary',
     'casbin',
     'sqlalchemy_adapter',
+    'passlib',
```

Comment:
The three dependencies here are only required for the server-side if user management is required for either SSO or basic auth. Can we avoid this dependencies by default and use `[server]` to let it install these dependencies?

Otherwise, this could increase the burden for a normal user to have additional dependencies.

Reply:
https://github.com/skypilot-org/skypilot/issues/6020